src/couch/src/couch_bt_engine_compactor.erl

```diff
@@ -58,6 +58,7 @@ start(#st{} = St, DbName, Options, Parent) ->
     NewSt3 = sort_meta_data(NewSt2),
     NewSt4 = commit_compaction_data(NewSt3),
     NewSt5 = copy_meta_data(NewSt4),
+    couch_db_plugin:before_copy_purge_info(DbName),
```

Comment:
This is an oddly specific place and name for a plugin's hook, in my opinion. The idea behind epi plugins that they could be used to extend common public functionality to allow to do various things. The `before_copy_purge_info` is  rather obscure code point and I don't imagine there are much could be done with it or even many devs would _know_ what to do with it. In my opinion the plugin hook should be in [start_compaction](https://github.com/apache/couchdb/blob/master/src/couch/src/couch_bt_engine.erl#L492) and named `before_compaction`.

Reply:
Does it have to be a plugin at all? How is a plugin better than a function call here?

Reply:
@nickva I think the idea was that this bit needed only during migration from pre-purge to post-purge and with plugin it'll be easier to take this out later. Sure it could be just a function and we'll remove code a release or two after, but presumably you can also do some interesting things with `before_compaction` hook, more intelligent compaction daemon with feedback, for example.

Reply:
Correct on @eiri's last point. This is only needed for doing the PSE upgrade. We did this as an EPI entry point so that we don't have to hard code knowledge about hastings and dreyfus since they're optional indexers that not everyone has deployed.